packages/discovery-react-components/src/components/DocumentPreview/components/PdfViewer/useAsyncFunctionCall.ts

```diff
@@ -0,0 +1,46 @@
+import { useEffect, useState } from 'react';
```

Comment:
This hook isn't specific to `PdfViewer`, so it should be moved up to the top-level `utils` dir.